sunpy/spectra/tests/test_spectrogram.py

```diff
@@ -14,6 +14,16 @@
     Spectrogram, LinearTimeSpectrogram, _LinearView
 )
 
+from sunpy.spectra.spectrum import Spectrum
+
+def test_spectrum():
+    #The idea is to create two 1-D arrays, data and freq_axis,
+    #Pass them to the Spectrum object and check for equality of dimensions accordingly.
+    data = np.linspace(0, 100, 100)
+    freq = np.linspace(1, 100, 100)
+    spec = Spectrum(data,freq)
+    if not (np.shape(spec.data) == np.shape(spec.freq_axis)):
+        raise ValueError
```

Comment:
1. Yes, that's correct. The tests of the code go in the test folders. But checking the data input is not a test of code, but a data validation should happen on initialisation
2. Yes
3. In the tests you can test whether the proper error is being raised. Check in this same file the [`test_rescale_error()`](https://github.com/sunpy/sunpy/blob/master/sunpy%2Fspectra%2Ftests%2Ftest_spectrogram.py#L502) function.